src/reservoir/utils.ts

```diff
@@ -0,0 +1,348 @@
+// @ts-ignore
+// eslint-disable-next-line import/no-unresolved
+import { adaptViemWallet } from "@reservoir0x/reservoir-sdk";
+import {
+  Address,
+  decodeFunctionData,
+  encodeAbiParameters,
+  encodeFunctionData,
+  Hash,
+  zeroAddress,
+} from "viem";
+
+import { Wallet, zeroHash } from "@/blockchain";
+import { seaportABI } from "@/generated/blockchain/seaport";
+
+export interface Offer {
+  itemType: number;
+  token: Address;
+  identifierOrCriteria: bigint;
+  startAmount: bigint;
+  endAmount: bigint;
+}
+
+export interface Consideration extends Offer {
+  recipient: Address;
+}
+
+export interface SeaportAskOrBid {
+  rawData: SeaportOrder;
+  price: {
+    netAmount: {
+      raw: string;
+    };
+  };
+}
+
+export interface SeaportOrder {
+  offerer: Address;
+  zone: Address;
+  offer: Offer[];
+  consideration: Consideration[];
+  orderType: number;
+  startTime: bigint;
+  endTime: bigint;
+  zoneHash: Hash;
+  salt: bigint;
+  conduitKey: Hash;
+  counter: bigint;
+}
+
+export interface SeaportOrderParameter extends SeaportOrder {
+  totalOriginalConsiderationItems: bigint;
+}
+
+export interface FulfillmentComponent {
+  orderIndex: bigint;
+  itemIndex: bigint;
+}
+
+export interface Fulfillment {
+  offerComponents: FulfillmentComponent[];
+  considerationComponents: FulfillmentComponent[];
+}
+
+export const ETH_CONTRACT_ADDRESS = zeroAddress;
+export const WETH_CONTRACT_ADDRESS =
+  "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+const SEAPORT_CONTRACT_NAME = "Seaport";
+const SEAPORT_VERSION = "1.5";
+const SEAPORT_CONTRACT_ADDRESS =
+  "0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC" as const;
+
+// We don't have this abi in the generated files
+export const EXECUTION_INFO_ABI = [
+  {
+    name: "ExecutionInfo",
+    type: "tuple",
+    components: [
+      { name: "module", type: "address" },
+      { name: "data", type: "bytes" },
+      { name: "value", type: "uint256" },
+    ],
+  },
+] as const;
```

Comment:
I'm kind of confused with all these. Some of it should be moved to the deploy get contracts file. Other data is about seaport, why having those here?